test/ecmascript.test.js

```diff
@@ -0,0 +1,64 @@
+const fs = require('fs');
+const path = require('path');
+const nodeFileTrace = require('../src/node-file-trace');
+const os = require('os');
+const { promisify } = require('util');
+const rimraf = require('rimraf');
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+
+const randomTmpId = Math.random().toString().slice(2);
+const tmpdir = path.resolve(os.tmpdir(), `node-file-trace-ecmascript${randomTmpId}`);
+rimraf.sync(tmpdir);
+fs.mkdirSync(tmpdir);
+console.log('created directory ' + tmpdir);
+
+ // These are tests known to fail so we skip them
+const ignoreCategories = new Set([
+  'optional catch binding',
```

Comment:
In a future PR, I will update `acorn` to see if it fixes "optional catch binding".

```
try {
  foo();
} catch {
  // nothing
}
```